src/module_5/module5.md

```diff
@@ -0,0 +1,5880 @@
+      <td>4.610000</td>
+    </tr>
+    <tr>
+      <th>35217</th>
+      <td>MS</td>
+      <td>0.456835</td>
+      <td>28.510000</td>
+    </tr>
+    <tr>
+      <th>35804</th>
+      <td>HIG</td>
+      <td>0.457259</td>
+      <td>11.870000</td>
+    </tr>
+    <tr>
+      <th>35166</th>
+      <td>UHT</td>
+      <td>0.457321</td>
+      <td>31.520000</td>
+    </tr>
+    <tr>
+      <th>35241</th>
+      <td>IEC</td>
+      <td>0.457656</td>
+      <td>3.840000</td>
+    </tr>
+    <tr>
+      <th>35148</th>
+      <td>WINT</td>
+      <td>0.458231</td>
+      <td>13356.000000</td>
+    </tr>
+    <tr>
+      <th>35069</th>
+      <td>HRTX</td>
+      <td>0.458231</td>
+      <td>18.800000</td>
+    </tr>
+    <tr>
+      <th>35156</th>
+      <td>OLED</td>
+      <td>0.458231</td>
+      <td>9.780000</td>
+    </tr>
+    <tr>
+      <th>35476</th>
+      <td>TGH</td>
+      <td>0.458347</td>
+      <td>11.490000</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+sv_top_ticker = get_shap_values("2009-06-30T00:00:00.000000000", "WINT")
+fig = plt.gcf()
+ax = plt.gca()
+ax.set_position([0.3, 0.1, 0.65, 0.8])
+shap.plots.waterfall(sv_top_ticker[0])
+```
+
+
+    
+![png](module5_files/module5_84_0.png)
+    
+
+
+
+```python
+shap.plots.beeswarm(sv)
+```
+
+
+    
+![png](module5_files/module5_85_0.png)
+    
+
+
+Indeed we see that the most penalizing feature was close_0. In the plot, the feature close_0 has a SHAP value of approximately -0.06. This value indicates that for the specific instance being analyzed, the close_0 feature contributes to decreasing the model's output by 0.06 units from the base value (average model output over the dataset). If the model's output is a probability, this would mean the presence or value of close_0 for this instance decreases the predicted probability of the positive class by 0.06.
+
+The reverse stock splitting in this case is leaking info to the past; i.e. if the company approved a stock split (bad signal) it means it didn't go under good circumstances and the price is being calculated backwards, so higher prices will be associated with potential future stock splits, hence company in not good economic position and performing bad against sp500.
+
+In this case, we were carrying an investigation to verify a different stock splitting event presented in class, but I wouldn't conclude that fast that there should be something wrong with the model and the importance given to close_0 value. If I had more time, I would like to research a bit and confirm if this might be also something related to the "regression to the mean" phenomenon, which could give us an idea on why lower stocks might have higher probabilities for surpassing sp500!
+
+One possible solution would be to have the adjusted-prices for stocks, so stock splitting would not affect actual prices, but we do not have that feature in the current dataset (maybe online), so we will just drop the suspicious columns.
```

Comment:
You mean to reverse the splitting? That would be a good idea, tbh, and I think I try it. However, I think at the end of the day the best is to delete price related features and normalize everything by the company market cap